src/main/kotlin/edu/ics499/team1/app/repositories/TripRepository.kt

```diff
@@ -1,8 +1,18 @@
 package edu.ics499.team1.app.repositories
 
-import edu.ics499.team1.app.domains.Trip
+import edu.ics499.team1.app.entities.TripEntity
+import edu.ics499.team1.app.entities.VehicleEntity
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Modifying
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
 
-interface TripRepository : JpaRepository<Trip, Long> {
-  
+// custom queries by https://www.baeldung.com/spring-data-jpa-modifying-annotation
+interface TripRepository : JpaRepository<TripEntity, Int> {
+
+    @Modifying
+    @Query("update TripEntity set mileage = :mileage where tripId = :tripId")
+    fun updateMileage(@Param("tripId") tripId: Int, @Param("mileage") mileage: Int)
```

Comment:
@Modifying is used when we need to make a custom modifying (update, delete, etc) query.

@Query will be for custom queries that we cannot create with a query with a derived jpa/spring data query as shown below

Reply:
The table will be the name of the entity, the fields (mileage and tripId will be the name of those fields in the entity.

:mileage is how the @Param will insert those values into the query. This is also sql injection safe.